evaluator/perception_online_evaluator/src/metrics_calculator.cpp

```diff
@@ -443,7 +448,34 @@ void MetricsCalculator::updateHistoryPath()
 
   for (const auto & [uuid, stamp_and_objects] : object_map_) {
     std::vector<Pose> history_path;
-    for (const auto & [stamp, object] : stamp_and_objects) {
+    for (auto it = stamp_and_objects.begin(); it != stamp_and_objects.end(); ++it) {
```

Comment:
Can you change the format like following:

```
for (const auto & [uuid, histry_path] : history_path_map) {
...
}
```

Reply:
```
const auto & [prev_stamp, prev_object] = *std::prev(it);
```
I used iterator to get previous elements